include/bluetooth/mesh/gen_plvl.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+/**
+ * @file
+ * @defgroup bt_mesh_gen_plvl Generic Power Level Models
+ * @{
+ * @brief API for the Generic Power Level models.
+ */
+
+#ifndef BT_MESH_GEN_PLVL_H__
+#define BT_MESH_GEN_PLVL_H__
+
+#include <bluetooth/mesh.h>
+#include <bluetooth/mesh/model_types.h>
+
+/** @cond INTERNAL_HIDDEN */
+#define GEN_PLVL_OP_LEVEL_GET		BT_MESH_MODEL_OP_2(0x82, 0x15)
+#define GEN_PLVL_OP_LEVEL_SET		BT_MESH_MODEL_OP_2(0x82, 0x16)
+#define GEN_PLVL_OP_LEVEL_SET_UNACK	BT_MESH_MODEL_OP_2(0x82, 0x17)
+#define GEN_PLVL_OP_LEVEL_STATUS	BT_MESH_MODEL_OP_2(0x82, 0x18)
+#define GEN_PLVL_OP_LAST_GET		BT_MESH_MODEL_OP_2(0x82, 0x19)
+#define GEN_PLVL_OP_LAST_STATUS		BT_MESH_MODEL_OP_2(0x82, 0x1A)
+#define GEN_PLVL_OP_DEFAULT_GET		BT_MESH_MODEL_OP_2(0x82, 0x1B)
+#define GEN_PLVL_OP_DEFAULT_STATUS	BT_MESH_MODEL_OP_2(0x82, 0x1C)
+#define GEN_PLVL_OP_RANGE_GET		BT_MESH_MODEL_OP_2(0x82, 0x1D)
+#define GEN_PLVL_OP_RANGE_STATUS	BT_MESH_MODEL_OP_2(0x82, 0x1E)
+#define GEN_PLVL_OP_DEFAULT_SET		BT_MESH_MODEL_OP_2(0x82, 0x1F)
+#define GEN_PLVL_OP_DEFAULT_SET_UNACK	BT_MESH_MODEL_OP_2(0x82, 0x20)
+#define GEN_PLVL_OP_RANGE_SET		BT_MESH_MODEL_OP_2(0x82, 0x21)
+#define GEN_PLVL_OP_RANGE_SET_UNACK	BT_MESH_MODEL_OP_2(0x82, 0x22)
+/** @endcond */
+
+struct gen_plvl_set {
+	u16_t power;
+	const struct model_transition *transition;
+};
+
+struct gen_plvl_status {
+	u16_t present; /**< Current Power Level. */
```

Comment:
rename to current? (present sounds like it should be a bool).

Reply:
I will suggest retaining `present` here and at other relevant places. This directly maps onto the message field of the same name and it is usually understood instantly by the readers who have seen the spec. It helps with retaining the consistency with the spec defined terms.

Reply:
@trond-snekvik Maybe switch back to present then? If it is a Spec defined name for the field?

Reply:
Yeah, IDK. I think this falls into the same old question of whether we expect users to read the model spec or not. From an API point of view, `current` makes more sense than `present`, which normally means "this entity is present". IMO the expectation that users will read the spec is a bit unrealistic, and can become an excuse for not making sensible APIs and good documentation. The point of these model implementations is to allow users to write Bluetooth Mesh applications without implementing the spec, therefore the spec shouldn't be on the required reading list.

I'd like to keep @joerchan's suggestion, as the `present` name clashes with the general ncs lingo, which should be our primary alignment target.